composer/utils/inference.py

```diff
@@ -0,0 +1,115 @@
+"""
+
+import copy
+import logging
+import os
+import tempfile
+from typing import Callable, Optional, Sequence, Union
+
+import torch
+
+from composer.models.base import ComposerModel
+from composer.utils import ObjectStore, dist
+from composer.utils.file_helpers import get_file, is_tar
+from composer.utils.iter_helpers import ensure_tuple
+
+log = logging.getLogger(__name__)
+
+__all__ = [
+    'export_for_inference',
+]
+
+
+def export_for_inference(
+    model: ComposerModel,
+    target_format: str,
+    store_path: str,
+    store_object_store: Optional[ObjectStore] = None,
+    sample_input: Optional[Union[torch.Tensor, Sequence[torch.Tensor]]] = None,
+    surgery_algs: Optional[Union[Callable, Sequence[Callable]]] = None,
+    transforms: Optional[Union[Callable, Sequence[Callable]]] = None,
+    load_path: Optional[str] = None,
+    load_object_store: Optional[ObjectStore] = None,
+) -> None:
+    """Export model for inference from a local file, URI, or cloud object store.
+
+    Args:
+        model (ComposerModel): An instance of ComposerModel. Please note that model is not modified inplace.
+            A copy of the model is made and export related transforms work on this copy.
+        target_format (str):  format to export to. For example, torchscript, onnx, etc.
+        store_path: (str): The path string for storing the exported model.
+        store_object_store (ObjectStore, optional): If the ``path`` is in an object store
+            (i.e. AWS S3 or Google Cloud Storage), an instance of
+            :class:`~.ObjectStore` which will be used
+            to store the exported model. Otherwise, if the store_path is a local filepath, set to ``None``.
+        sample_input (Union[torch.Tensor, Sequence[torch.Tensor]], optional): Example model inputs used for tracing.
+        surgery_algs (Union[Callable, Sequence[Callable]], optional): Algorithms that should be applied to the model
+            before loading a checkpoint. Each should be callable that takes a model and returns modified model.
+        surgery_algs (Union[Callable, Sequence[Callable]], optional): transformations (usually optimizations) that should be applied to the model
+            before loading a checkpoint. Each should be a callable that takes a model and returns a modified model.
+        load_path (str): The path string to an existing checkpoint file.
+            It can be a path to a file on the local disk, a URL, or if ``load_object_store`` is set, the object name
+            for a checkpoint in a cloud bucket. For example, my_model/ep1-rank0.tar
+        load_object_store (ObjectStore, optional): If the ``path`` is in an object store
+            (i.e. AWS S3 or Google Cloud Storage), an instance of
+            :class:`~.ObjectStore` which will be used to retreive the checkpoint.
+            Otherwise, if the checkpoint is a local filepath, set to ``None``. (default: ``None``)
```

Comment:
Ideally we should be consistent about adding (default: ``whatever``) to either all args or none of them. (We used to do all, but maybe we stopped that?)

Reply:
I missed adding default. Will add for all of them.

Reply:
Added defaults.